### PR TITLE
Electra state changes

### PR DESCRIPTION
**Pull request by rolfyone**
Added the required state fields for EIP-7251 (MaxEB)
Some of the initializations are zeroed but need to be coming from state functions - these will be finalized in a follow-up PR, they need to be consistent with other implementations otherwise we'll be on different state roots, but equally this is 'ok' for now.

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/state/versions/electra/PendingPartialWithdrawal.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    public SszUInt64 getIndexSchema() {
+      return (SszUInt64) getFieldSchema0();
+    }
+
+    public SszUInt64 getAmountSchema() {
+      return (SszUInt64) getFieldSchema1();
+    }
+
+    public SszUInt64 getEpochSchema() {
```
Comment (mehdi-aouadi) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/state/versions/electra/PendingPartialWithdrawal.java]: Same here `getWithdrawalEpoch`

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/state/versions/electra/PendingPartialWithdrawal.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    extends Container3<PendingPartialWithdrawal, SszUInt64, SszUInt64, SszUInt64> {
+
+  public static class PendingPartialWithdrawalSchema
+      extends ContainerSchema3<PendingPartialWithdrawal, SszUInt64, SszUInt64, SszUInt64> {
+    public PendingPartialWithdrawalSchema() {
+      super(
+          "PendingPartialWithdrawal",
+          namedSchema("index", SszPrimitiveSchemas.UINT64_SCHEMA),
+          namedSchema("amount", SszPrimitiveSchemas.UINT64_SCHEMA),
+          namedSchema("epoch", SszPrimitiveSchemas.UINT64_SCHEMA));
```
Comment (mehdi-aouadi) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/state/versions/electra/PendingPartialWithdrawal.java]: This should be `withdrawable_epoch` instead [according to the spec](https://github.com/ethereum/consensus-specs/blob/88f5dde0f973151e9df84ba6b15d1c67287df7ee/specs/_features/eip7251/beacon-chain.md?plain=1#L182)

File: data/beaconrestapi/src/integration-test/resources/tech/pegasys/teku/beaconrestapi/beacon/schema/PendingPartialWithdrawal.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "title" : "PendingPartialWithdrawal",
+  "type" : "object",
+  "required" : [ "index", "amount", "epoch" ],
```
Comment (mehdi-aouadi) [inline data/beaconrestapi/src/integration-test/resources/tech/pegasys/teku/beaconrestapi/beacon/schema/PendingPartialWithdrawal.json]: Same here should be `withdrawable_epoch`

File: data/serializer/src/main/java/tech/pegasys/teku/api/schema/electra/PendingPartialWithdrawal.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+public class PendingPartialWithdrawal {
+  @JsonProperty("index")
+  public final int index;
+
+  @JsonProperty("amount")
+  public final UInt64 amount;
+
+  @JsonProperty("withdrawable_epoch")
+  public final UInt64 epoch;
```
Comment (mehdi-aouadi) [inline data/serializer/src/main/java/tech/pegasys/teku/api/schema/electra/PendingPartialWithdrawal.java]: Could rename the variable accordingly: `withdrawble_epoch`

**Review by mehdi-aouadi — APPROVED** [review]
LGTM
Just a small variable renaming needed (`withdrawable_epoch` instead of `epoch`)

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/state/versions/electra/PendingPartialWithdrawal.java (hunk 1 above)
Comment (rolfyone) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/state/versions/electra/PendingPartialWithdrawal.java]: i'll have to check the spec, it looked like it was called 'epoch' but maybe its been updated again... thanks!

Comment (rolfyone) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/state/versions/electra/PendingPartialWithdrawal.java]: it was always called that i think i just missed a rename in copying things, so appreciate the catch, one less thing to fix when implementing :)

**Pull request merged**